tdrs-backend/tdpservice/parsers/validators.py

```diff
@@ -23,276 +24,361 @@ def validator(value):
+        else (
+            False,
+            "".join(
+                [
+                    " and " + validator(value)[1] if validator(value)[0] else ""
+                    for validator in args
+                ]
+            ),
+        )
+    )
 
 
-def if_then_validator(condition_field, condition_function,
-                      result_field, result_function):
+def if_then_validator(
+    condition_field, condition_function, result_field, result_function
+):
     """Return second validation if the first validator is true.
 
     :param condition_field: function that returns (bool, string) to represent validation state
     :param condition_function: function that returns (bool, string) to represent validation state
     :param result_field: function that returns (bool, string) to represent validation state
     :param result_function: function that returns (bool, string) to represent validation state
     """
+
     def if_then_validator_func(value):
-        value1 = value[condition_field] if type(value) is dict else getattr(value, condition_field)
-        value2 = value[result_field] if type(value) is dict else getattr(value, result_field)
+        value1 = (
+            value[condition_field]
+            if type(value) is dict
+            else getattr(value, condition_field)
+        )
+        value2 = (
+            value[result_field] if type(value) is dict else getattr(value, result_field)
+        )
 
         validator1_result = condition_function(value1)
         validator2_result = result_function(value2)
-        return (True, None) if not validator1_result[0] else (
-            validator2_result[0], (f'if {condition_field} '
-                                   + (validator1_result[1] if validator1_result[1] is not None
-                                      else f":{value1} validator1 passed")
-                                   + f' then {result_field} '
-                                   + (validator2_result[1] if validator2_result[1] is not None
-                                      else "validator2 passed")) if not validator2_result[0] else None)
+        return (
+            (True, None, [condition_field, result_field])
+            if not validator1_result[0]
+            else (
+                validator2_result[0],
+                (
+                    f"if {condition_field} "
+                    + (
+                        validator1_result[1]
+                        if validator1_result[1] is not None
+                        else f":{value1} validator1 passed"
+                    )
+                    + f" then {result_field} "
+                    + (
+                        validator2_result[1]
+                        if validator2_result[1] is not None
+                        else "validator2 passed"
+                    )
+                )
+                if not validator2_result[0]
+                else None,
+                [condition_field, result_field],
+            )
+        )
```

Comment:
Please consider refactoring this for cleaner more pythonic style and perhaps using an accessory function to simplify.